triton-vm/src/table/table_collection.rs

```diff
@@ -481,9 +482,12 @@ impl ExtTableCollection {
                 if let Some(profiler) = maybe_profiler.as_mut() {
                     profiler.start(&ext_codeword_table.name());
                 }
+                // TODO: Consider if we can use `transposed_ext_codewords` from caller, Stark::prove().
+                // This would require more complicated indexing, but it would save a lot of allocation.
+                let transposed_codewords = Stark::transpose(ext_codeword_table.data());
```

Comment:
I feel like it's cleaner if `TableCollection` does not depend on `Stark`. Since you anticipate passing the transposed codewords from `Stark::prove()` directly, I interpret this line as a temporary smell.

Reply:
the `transpose()` function should live in twenty-first.